packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTableSet.tsx

```diff
@@ -0,0 +1,281 @@
+                index !== 0 ? this._renderReferencesTable(item) : undefined
+              )}
+            </Stack>
+          )}
+        </Stack>
+      );
+    }
+    return undefined;
+  }
+
+  private _renderReferencesTable(item: IApiProperty): JSX.Element {
+    return (
+      <ApiReferencesTable
+        key={item.propertyName}
+        name={item.propertyName}
+        title={item.title}
+        description={item.description}
+        extendsTokens={item.extendsTokens}
+        properties={item.property}
+        methods={item.methods}
+        renderAsEnum={item.propertyType === PropertyType.enum}
+        renderAsClass={item.propertyType === PropertyType.class}
+      />
+    );
+  }
+
+  private _onHashChange = (): void => {
+    const { properties, showSeeMore } = this.state;
+
+    const hash = extractAnchorLink(window.location.hash);
+    const el = document.getElementById(hash);
+
+    if (el) {
+      // update scroll position
+      window.scrollTo({ top: el.offsetTop - TITLE_LINE_HEIGHT /* title */, behavior: 'smooth' });
+    }
+
+    if (!showSeeMore) {
+      const section = properties.filter(x => x.propertyName === hash)[0];
+      if (section) {
+        this.setState({
+          showSeeMore: true
+        });
+      }
+    }
+  };
+
+  private _onRenderText(): JSX.Element {
+    return <Text variant="xLarge">See more</Text>;
+  }
+
+  private _onClickSeeMore = (): void => {
+    this.setState({
+      showSeeMore: !this.state.showSeeMore
+    });
+  };
+
+  private _generatePropertyArray(): Array<IApiProperty> {
+    const iComponentProps: Array<IApiProperty> = [];
+    const preResults: Array<IApiProperty> = [];
+
+    const { jsonDocs } = this.props;
+
+    if (jsonDocs) {
+      const propsName: string = `I${jsonDocs.name}Props`;
+      for (let j = 0; j < jsonDocs.tables.length; j++) {
+        switch (jsonDocs.tables[j].kind) {
+          case 'enum': {
+            preResults.push(this._generateEnumProperty(jsonDocs.tables[j]));
+            break;
+          }
+          case 'interface': {
+            const interfaceProperty = this._generateInterfaceProperty(jsonDocs.tables[j]);
+
+            // to ensure that I{componentName}Props comes first
+            if (propsName === jsonDocs.tables[j].name) {
+              iComponentProps.push(interfaceProperty);
+            } else {
+              preResults.push(interfaceProperty);
+            }
+
+            break;
+          }
+          case 'class': {
+            preResults.push(this._generateClassProperty(jsonDocs.tables[j]));
+            break;
+          }
+        }
+      }
+    }
+
+    return [...iComponentProps, ...preResults];
+  }
+
+  private _generateEnumProperty(table: ITableJson): IApiProperty {
+    const enumMembers: IApiEnumProperty[] = [];
+
+    const members: IEnumTableRowJson[] = table.members as IEnumTableRowJson[];
+    for (let k = 0; k < members.length; k++) {
```

Comment:
nit, I suggest using `map` here (and any time there's a 1-1 mapping from items of the original array to a new array)